### PR TITLE
Replace actorRef2Scala with value class.

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: akka-actor/src/main/mima-filters/2.6.5.backwards.excludes/ActorRefOps.excludes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Replace actorRef2Scala with value class.
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.package.actorRef2Scala")
```
Comment (johanandren) [inline akka-actor/src/main/mima-filters/2.6.5.backwards.excludes/ActorRefOps.excludes:2]: I pretty sure that's not safe to filter, any code using `!` compiled against 2.6.4- would stop working.

Comment (hepin1989) [reply akka-actor/src/main/mima-filters/2.6.5.backwards.excludes/ActorRefOps.excludes:2]: Yes... Missed that.

**Comment by hepin1989** [comment]
Thins can be done once BC is available

File: akka-actor/src/main/scala/akka/actor/package.scala [hunk 2]
```diff
@@ -7,6 +7,5 @@ package akka
 import language.implicitConversions
 
 package object actor {
-  @inline implicit final def actorRef2Scala(ref: ActorRef): ScalaActorRef = ref.asInstanceOf[ScalaActorRef]
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/package.scala:10]: at least there is no performance overhead of this

**Pull request closed without merging**